packages/generator/src/prisma/model.ts

```diff
@@ -0,0 +1,77 @@
+import {Field, FieldType} from './field'
+import {singlePascal} from '../utils/plurals'
+
+function stringifyFieldsForPrinting(fields: Field[]) {
+  let maxNameLength = 1
+  let maxTypeLength = 1
+  for (let field of fields) {
+    const [name, type] = field.toString().split(/[\s]+/)
+    maxNameLength = Math.max(maxNameLength, name.length)
+    maxTypeLength = Math.max(maxTypeLength, type.length)
+  }
+  // add horizontal padding, otherwise there would be no space after the type
+  maxNameLength += 2
+  maxTypeLength += 2
+  return fields.map((field) => {
+    const [name, type, ...rest] = field.toString().split(/[\s]+/)
+    const attrs = rest.join(' ')
+    const namePad = maxNameLength - name.length
+    const typePad = maxTypeLength - type.length
+    return `${name}${Array(namePad).join(' ')}${type}${Array(typePad).join(' ')}${attrs}`
+  })
+}
+
+export class Model {
+  name: string
+  fields: Field[]
+
+  constructor(name: string, fields: Field[] = []) {
+    this.name = singlePascal(name)
+    this.fields = fields
+  }
+
+  private getIdField() {
+    return new Field('id', {
+      isRequired: true,
+      isList: false,
+      isId: true,
+      default: 'autoincrement()',
+      type: FieldType.Int,
+    })
+  }
+
+  private getCreatedAtField() {
+    return new Field('createdAt', {
+      isRequired: true,
+      isList: false,
+      isId: false,
+      default: 'now()',
+      type: FieldType.DateTime,
+    })
+  }
+
+  private getUpdatedAtField() {
+    return new Field('updatedAt', {
+      isRequired: true,
+      isList: false,
+      isId: false,
+      isUpdatedAt: true,
+      type: FieldType.DateTime,
+    })
+  }
+
+  private getFields() {
+    return stringifyFieldsForPrinting([
+      this.getIdField(),
+      ...this.fields,
+      this.getCreatedAtField(),
+      this.getUpdatedAtField(),
```

Comment:
What do you think about leaving id, createdAt, and updatedAt all at the beginning? I like this because then all the "automatic" ones are grouped together. And later when I come back to add a new field, I can just add it to the end. This would also help if we support adding fields via CLI.

```suggestion
      this.getCreatedAtField(),
      this.getUpdatedAtField(),
      ...this.fields,
```

Reply:
I thought about this, my concern was that it makes the studio a little more tedious to use - columns are ordered, so it led to the first 3 columns in the DB being `id`, `createdAt`, `updatedAt`, which means most of the UI you see isn't super useful. I think from a technical perspective it doesn't matter too much, but there's maybe some usability concerns there

Reply:
Yeah that's a good point. I pondered this for a bit. And I think I specifically like the dates at the beginning in a UI. Often the dates are one of the items you are looking at or sorting by.

Reply:
Interesting, I felt the opposite way but totally get the argument that you might want to sort by date. I'm gonna drop a quick poll in the slack and see what folks think